### PR TITLE
Feature/android increase batch timeout leeway

### PR DESCRIPTION
**Pull request by Dartlexx**

### CODE REVIEW

File: vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt [hunk 1]
```diff
@@ -29,7 +30,7 @@ class AndroidDeviceTestRunner(private val device: AndroidDevice) {
     companion object {
         private val JUNIT_IGNORE_META_PROPERTY = MetaProperty("org.junit.Ignore")
         private const val MAX_TEST_DURATION_LEEWAY = 1.2 // +20% to expected test duration
-        private const val BATCH_DURATION_LEEWAY = 1.1 // +10% to expected batch duration
+        private const val BATCH_DURATION_LEEWAY = 1.3 // +30% to expected batch duration
```
Comment (memega) [inline vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt:33]: These _almost_ look like something that should be defined in configuration file.

Comment (Dartlexx) [reply vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt:33]: And it is. Actually, my solution is that timeout is based on the biggest value - either from historical data (combined for batch) or from configuration.

Comment (Dartlexx) [reply vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt:33]: I just increased the batch timeout in Marathon config (different repo).

**Review by memega — APPROVED** [review]

File: vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt [hunk 2]
```diff
@@ -115,7 +116,7 @@ class AndroidDeviceTestRunner(private val device: AndroidDevice) {
             max(configuration.testOutputTimeoutMillis, (testBatch.maxExpectedTestDurationMs * MAX_TEST_DURATION_LEEWAY).toLong())
         )
 
-        // Batch duration timeout can't be less then 'testBatchTimeoutMillis'. There are also additional 10% for unexpected cases.
+        // Batch duration timeout can't be less then 'testBatchTimeoutMillis'. There are also additional 30% for unexpected cases.
```
Comment (Malinskiy) [inline vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt:119]: Keep in mind that the timeout is not fully respected by ddmlib: it is only taken into account for some part of the test execution command. If you want to have a reliable timeout management - consider merging with latest develop and try out the adam vendor module. Adam has full transparency in terms of timout management: no concurrent threads executed and locks/latches.

**Review by Malinskiy — COMMENTED** [review]
FYI

**Pull request merged**